contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "../../issuance/IssuanceEth.sol";
+import "../token/ERC20MultiDividendable.sol";
+
+
+contract DAO is ERC20Mintable, ERC20MultiDividendable, IssuanceEth {
+
+    mapping(address => uint256) public begAmount;
+    uint256 totalAmounts;
+    mapping(address =>
+        mapping(address => uint256)
+    ) public votesForIdeaByHolder;
+    mapping(address => uint256) public totalVotesForIdea;
+    mapping(address => uint256) public totalVotesByHolder;
+    mapping(address => address[]) public backersForIdea;
+    event Here();
+
+    constructor()
+    ERC20Mintable()
+    ERC20MultiDividendable()
+    IssuanceEth(address(this))
+    public
+    {
+        _createState("NEVER");
+        _createTransition("LIVE", "SETUP");
+        _createTransition("FAILED", "SETUP");
+        addMinter(address(this));
+    }
+
+    function transferFunds(address payable _wallet) public onlyOwner {
+        require(currentState == "NEVER", "You can never do this.");
+    }
+
+    function restartFundingRound() public onlyOwner {
+        require(
+            currentState == "LIVE" || currentState == "FAILED",
+            "Initial funding round not ended."
+        );
+        _transition("SETUP");
+    }
+
+    function begMoneyForIdea(uint256 amount, address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalAmounts + amount <= address(this).balance,
+            "You beg too much."
+        );
+        begAmount[idea] = amount;
+    }
+
+    function voteForIdea(uint256 votes, address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            this.balanceOf(msg.sender) - totalVotesByHolder[msg.sender] >= votes,
+            "Not enough power."
+        );
+        totalVotesForIdea[idea] += votes;
+        totalVotesByHolder[msg.sender] += votes;
+        votesForIdeaByHolder[idea][msg.sender] += votes;
+        resolveBackerForIdea(idea, msg.sender);
+    }
+
+    function fundIdea(address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalVotesForIdea[idea] >= this.totalSupply().div(2) + 1,
+            "Not enough expressed votes."
+        );
+        // solium-disable-next-line security/no-call-value
+        IssuanceEth(idea).invest.value(begAmount[idea])();
```

Comment:
Speaking of reentrancy and DAO's https://ethernaut.openzeppelin.com/level/0xf70706db003e94cfe4b5e27ffd891d5c81b39488
Also, it might be interesting to have a look at https://consensys.github.io/smart-contract-best-practices/recommendations/#handle-errors-in-external-calls